src/compiler/core.ts

```diff
@@ -2261,70 +2256,6 @@ const createUIStringComparer = (() => {
         const comparer = new Intl.Collator(locale, { usage: "sort", sensitivity: "variant" }).compare;
         return (a, b) => compareWithCallback(a, b, comparer);

```

Comment:
Another cleanup here would be to remove this fallback; IIRC you can get all of the way there with only collator options (though this fallback method is not uncommon; it's what eslint-plugin-simple-import-sort does).